components/jupyter-web-app/backend/kubeflow_jupyter/common/utils.py

```diff
@@ -269,37 +279,57 @@ def process_resource(rsrc):
     return res
 
 
-def process_status(rsrc):
+def process_status(rsrc, rsrc_events):
     '''
     Return status and reason. Status may be [running|waiting|warning|error]
     '''
     # If the Notebook is being deleted, the status will be waiting
     if "deletionTimestamp" in rsrc["metadata"]:
-        return "waiting", "Deleting Notebook Server"
+        return STATUS_WAITING, "Deleting Notebook Server"
 
     # Check the status
     try:
         s = rsrc["status"]["containerState"]
     except KeyError:
-        return "waiting", "No Status available"
+        s = ""
```

Comment:
Could you add a small comment explaining that if no `containerState` is available, then we provide information regarding the Notebook's state from the Events in the if clause bellow?

Reply:
Done